fr.esrf.tango.generator.xtend/src/fr/esrf/tango/pogo/generator/cpp/utils/Attributes.xtend

```diff
@@ -267,25 +281,25 @@ class Attributes {
 	def Constructor(Attribute attribute, boolean dynamic) '''
 		«IF dynamic»
 			«IF attribute.isScalar»
-				«attribute.name»Attrib(const std::string &att_name):«attribute.inheritance»(att_name.c_str(), 
+				«attribute.name»Attrib(const std::string &att_name):«attribute.inheritance»(att_name.c_str(),
 						«attribute.dataType.cppTypeEnum», Tango::«attribute.rwType») {};
 			«ELSEIF attribute.isSpectrum»
-				«attribute.name»Attrib(const std::string &att_name):«attribute.inheritance»(att_name.c_str(), 
-						«attribute.dataType.cppTypeEnum», Tango::«attribute.rwType», «attribute.maxX») {};
+				«attribute.name»Attrib(const std::string &att_name):«attribute.inheritance»(att_name.c_str(),
+						«attribute.dataType.cppTypeEnum», Tango::«attribute.rwType», «attribute.name»_X_DATA_SIZE) {};
 			«ELSE»
-				«attribute.name»Attrib(const std::string &att_name):«attribute.inheritance»(att_name.c_str(), 
-						«attribute.dataType.cppTypeEnum», Tango::«attribute.rwType», «attribute.maxX», «attribute.maxY») {};
+				«attribute.name»Attrib(const std::string &att_name):«attribute.inheritance»(att_name.c_str(),
+						«attribute.dataType.cppTypeEnum», Tango::«attribute.rwType», «attribute.name»_X_DATA_SIZE, «attribute.name»_Y_DATA_SIZE) {};
 			«ENDIF»
 		«ELSE»
 			«IF attribute.isScalar»
 				«attribute.name»Attrib():«attribute.inheritance»("«attribute.name»",
 						«attribute.dataType.cppTypeEnum», Tango::«attribute.rwType»«attribute.readWithWrite») {};
 			«ELSEIF attribute.isSpectrum»
 				«attribute.name»Attrib():«attribute.inheritance»("«attribute.name»",
-						«attribute.dataType.cppTypeEnum», Tango::«attribute.rwType», «attribute.maxX») {};
+						«attribute.dataType.cppTypeEnum», Tango::«attribute.rwType», «attribute.name»_X_DATA_SIZE) {};
 			«ELSE»
 				«attribute.name»Attrib():«attribute.inheritance»("«attribute.name»",
-						«attribute.dataType.cppTypeEnum», Tango::«attribute.rwType», «attribute.maxX», «attribute.maxY») {};
+						«attribute.dataType.cppTypeEnum», Tango::«attribute.rwType», «attribute.name»_X_DATA_SIZE, «attribute.name»_Y_DATA_SIZE) {};
 			«ENDIF»
 		«ENDIF»
```

Comment:
Some more changes are needed in cpp generated code. Still these sizes are hardcoded in some places:
```cpp
	/*----- PROTECTED REGION ID(Mytestclass::read_myimg) ENABLED START -----*/
	//	Set the attribute value
	attr.set_value(attr_myimg_read, 100, 200);

	/*----- PROTECTED REGION END -----*/	//	Mytestclass::read_myimg
```

I'd expect to see instead:
```cpp
attr.set_value(attr_myimg_read, myimgAttr::X_DATA_SIZE, myimgAttr::Y_DATA_SIZE);
```

Reply:
Nice catch, thx!
Now we have:
```cpp
//--------------------------------------------------------
/**
 *	Read attribute spectrumDynamicTest related method
 *	Description: 
 *
 *	Data type:	Tango::DevDouble
 *	Attr type:	Spectrum max = 10
 */
//--------------------------------------------------------
void Test::read_spectrumDynamicTest(Tango::Attribute &attr)
{
	DEBUG_STREAM << "Test::read_spectrumDynamicTest(Tango::Attribute &attr) entering... " << std::endl;
	Tango::DevDouble	*att_value = get_spectrumDynamicTest_data_ptr(attr.get_name());
	/*----- PROTECTED REGION ID(Test::read_spectrumDynamicTest) ENABLED START -----*/
	//	Set the attribute value
	attr.set_value(att_value, spectrumDynamicTestAttrib::X_DATA_SIZE);
	
	/*----- PROTECTED REGION END -----*/	//	Test::read_spectrumDynamicTest
}
```